### PR TITLE
RTDB v2 Triggers

### PR DESCRIPTION
**Pull request by colerogers**
Adds sample for `uppercase` using v2 functions.

### CODE REVIEW

File: 2nd-gen/uppercase/functions/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "deploy": "firebase deploy --only functions",
+    "logs": "firebase functions:log"
+  },
+  "engines": {
+    "node": "16"
+  },
+  "main": "index.js",
+  "dependencies": {
+    "firebase-admin": "^10.0.2",
+    "firebase-functions": "^4.0.0"
```
Comment (jhuleatt) [inline 2nd-gen/uppercase/functions/package.json:18]:
```suggestion
    "firebase-functions": "^3.22.0"
```

We can move the whole repo to v4 when it's ready

**Review by jhuleatt — CHANGES_REQUESTED** [review]

File: 2nd-gen/uppercase/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Firebase SDK for Cloud Functions 2nd Gen Quickstart - Realtime Database
+
+This quickstart demonstrates using **Firebase SDK for Cloud Functions** setup with a **Firebase database**.
```
Comment (markarndt) [inline 2nd-gen/uppercase/README.md:3]: Suggested "using **Firebase SDK for Cloud Functions** with with **Firebase Realtime Database**.

File: 2nd-gen/uppercase/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+## Introduction
+
+This sample app does two things:
+ - Create messages in the Firebase Realtime Database using a simple HTTPS request which is handled by a v2 HTTPS Firebase Function. Writing to the Realtime Database is done using the Firebase Admin SDK.
```
Comment (markarndt) [inline 2nd-gen/uppercase/README.md:9]: Please check with egilmorez, but here the usage is closer to "...a v2 HTTPS function."

File: 2nd-gen/uppercase/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+This quickstart demonstrates using **Firebase SDK for Cloud Functions** setup with a **Firebase database**.
+
+
+## Introduction
+
+This sample app does two things:
+ - Create messages in the Firebase Realtime Database using a simple HTTPS request which is handled by a v2 HTTPS Firebase Function. Writing to the Realtime Database is done using the Firebase Admin SDK. 
+ - When a message gets added in the Realtime Database, a v2 Firebase Function triggers and automatically makes these messages all uppercase.
+
```
Comment (markarndt) [inline 2nd-gen/uppercase/README.md:11]: Check with egilmorez... "...a v2 function triggers and...."

File: 2nd-gen/uppercase/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+## Deploy and try out
+
+To deploy and try out the sample:
+
+ - Create a Firebase project on the [Firebase Console](https://console.firebase.google.com)
```
Comment (markarndt) [inline 2nd-gen/uppercase/README.md:16]: "...project using the..."

Comment (markarndt) [inline 2nd-gen/uppercase/README.md:16]: "Firebase console"

File: 2nd-gen/uppercase/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+ - Install the required dependencies by running `npm install` in the `functions` directory
+ - Deploy your project's code using `firebase deploy`
+ - Create a message by opening the querying the functions url from your browser.
```
Comment (markarndt) [inline 2nd-gen/uppercase/README.md:19]: "URL"

File: 2nd-gen/uppercase/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+ The function executes and redirects the browser to the Firebase console at the database location where the text string was stored. You should see your text value displayed in the console and uppercase.
```
Comment (markarndt) [inline 2nd-gen/uppercase/README.md:21]: "in uppercase"

File: 2nd-gen/uppercase/README.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+## Contributing
+
+We'd love that you contribute to the project. Before doing so please read our [Contributor guide](../../CONTRIBUTING.md).
```
Comment (markarndt) [inline 2nd-gen/uppercase/README.md:25]: "We'd love you to contribute...."

File: 2nd-gen/uppercase/functions/index.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+// [START makeuppercase]
+// Listens for new messages added to /messages/:pushId/original and creates an
+// uppercase version of the message to /messages/:pushId/uppercase
+// for all databases in 'us-central1'
+exports.makeuppercase = onValueCreated('/messages/{pushId}/original', (event) => {
+      // Grab the current value of what was written to the Realtime Database.
+      const original = event.data.val();
+      logger.log('Uppercasing', event.params.pushId, original);
+      const uppercase = original.toUpperCase();
+      // You must return a Promise when performing asynchronous tasks inside a Functions such as
```
Comment (markarndt) [inline 2nd-gen/uppercase/functions/index.js:56]: This error comes from the published docs, of course. Let's correct here to read "...tasks inside a function, such as..."

**Review by markarndt — APPROVED** [review]
Hopefully my individual comments are preserved after this review is submitted....

File: 2nd-gen/uppercase/functions/index.js (hunk 9 above)
Comment (colerogers) [reply 2nd-gen/uppercase/functions/index.js]: 👍

File: 2nd-gen/uppercase/README.md (hunk 7 above)
Comment (colerogers) [reply 2nd-gen/uppercase/README.md]: 👍

File: 2nd-gen/uppercase/README.md (hunk 6 above)
Comment (colerogers) [reply 2nd-gen/uppercase/README.md]: 👍

File: 2nd-gen/uppercase/README.md (hunk 5 above)
Comment (colerogers) [reply 2nd-gen/uppercase/README.md]: 👍

Comment (colerogers) [reply 2nd-gen/uppercase/README.md]: 👍

File: 2nd-gen/uppercase/README.md (hunk 4 above)
Comment (colerogers) [reply 2nd-gen/uppercase/README.md:11]: 👍

File: 2nd-gen/uppercase/README.md (hunk 3 above)
Comment (colerogers) [reply 2nd-gen/uppercase/README.md]: 👍

**Pull request merged**